.github/workflows/examples-nxp.yaml

```diff
@@ -208,6 +208,140 @@ jobs:
               if: ${{ !env.ACT }}
               with:
                 platform-name: MCXW71
+    rt1060:
```

Comment:
This adds a lot of builds. We already have k32w0, k32w1, mcxw71, rw61x and zephyr.

Can we trim down the number of examples or have some common items or any other strategy to not just have a combinatorial explosion of all supported hardware?

Reply:
Our RT platforms support different host/tranceiver configuration, so we want to make sure to cover the build for all those configuration.
We also need to build each example at least once for those platform to make sure other changes in matter are tracked and fixed by other contributors.
What we can do it to spread those multiple configuration among the different examples, meaning we would build a config only once, for one of our example.
For instance:
```

            - name: Build RT10600 all clusters example app
              run: |
                  scripts/run_in_build_env.sh "\
                      ./scripts/build/build_examples.py \
                      --target nxp-rt1060-freertos-all-clusters-wifi-iw416 \
                      --target nxp-rt1060-freertos-all-clusters-wifi-w8801 \
                      build \
                      --copy-artifacts-to out/artifacts \
                  "
            - name: Build RT1060 thermostat example app
              run: |
                  scripts/run_in_build_env.sh "\
                      ./scripts/build/build_examples.py \
                      --target nxp-rt1060-freertos-thermostat-thread-wifi-evkc-iwx12 \
                      build \
                      --copy-artifacts-to out/artifacts \
                  "
            - name: Build RT1060 laundry-washer example app
              run: |
                  scripts/run_in_build_env.sh "\
                      ./scripts/build/build_examples.py \
                      --target nxp-rt1060-freertos-laundry-washer-wifi-evkc-iwx12 \
                      --target nxp-rt1060-freertos-laundry-washer-thread-evkc-iwx12 \
                      build \
                      --copy-artifacts-to out/artifacts \
                  "
```

This would allow us to cover all supported examples, and all our host/tranceiver config at least once.
What do you think ?

Reply:
NXP currently uses almost 3 hours of combined CI time

```
Build example - NXP / RW61X                                                                             2:43:14              0:58:01
Build example - NXP / ZEPHYR                                                                            2:43:14              0:34:11
Build example - NXP / K32W0                                                                             2:43:14              0:31:50
Build example - NXP / K32W1                                                                             2:43:14              0:19:45
Build example - NXP / MCXW71                                                                            2:43:14              0:19:27
```

please trim these down, ideally under 45 minutes. Splitting across jobs tries to hide the problem, however the total agregate time still exists.

Reply:
I think the question is if we really believe that board differences will significantly change build results. Zephyr vs FreeRTOS is probably very different. Beyond that I do not know. I would generally maybe consider one sample app per board and also merge these into a single workflow so as to not pay the cost of checkout & bootstrap on each (that is about 6 minutes each time)

Reply:
Number of examples is now reduced with keeping minimum build coverage.

Reply:
This particular PR right now takes 2.5 hours:

![image](https://github.com/user-attachments/assets/218ab341-651e-4bf7-8dec-e0fd8a88a894)

Reply:
almost 2 hours to check that "things build" seems a bit too much. 

![image](https://github.com/user-attachments/assets/c585107d-d73b-48d9-8744-7503c9418e07)

Reply:
Please update your flags to only compile a subset for PRs and let master merges compile the 2hr CI.

This is what is done in other CI runs: https://github.com/project-chip/connectedhomeip/blob/master/.github/workflows/examples-nrfconnect.yaml#L80

Reply:
Just added a commit with PR filter
Thanks for guidance

Reply:
@andy31415 are you ok with latest changes ?